build-bin/maven/maven_unjar

```diff
@@ -0,0 +1,96 @@
+#!/bin/sh
+#
+# Copyright 2019-2020 The OpenZipkin Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+#
+
+set -eu
+
+# This script gets one jar from Maven, most typically an exec or module jar, extracting its contents
+# to a directory corresponding to the artifact_id parameter.
+#
+# Ex. ./maven_unjar io.zipkin zipkin-server 2.22.2 slim
+#
+# When the version ends in SNAPSHOT, this searches the current directory instead of resolving
+# remotely. If the intended artifact isn't there, it is built on-demand using `mvn package`.
+#
+# Note: Be careful building on-demand within Docker, as it can cause high bandwidth usage, pulling
+# dependencies for the project. If you are running this within Docker, ensure you have .dockerignore
+# setup properly to pass the artifact you are looking for.
+#
+# Ex.
+# !./module/target/zipkin-module-storage-kafka-*-module.jar
+
+group_id=${1?group_id is required}
+artifact_id=${2?artifact_id is required}
+version=${3?version is required}
+classifier=${4:-}
+
+if [ -n "${classifier}" ]; then
+  qualified_jar=${artifact_id}-${version}-${classifier}.jar
+else
+  qualified_jar=${artifact_id}-${version}.jar
+fi
+
+case ${version} in
+  *-SNAPSHOT )
+    is_snapshot=true
+    ;;
+  * )
+    is_snapshot=false
+    ;;
+esac
+
+local_repo=$HOME/.m2/repository
+local_repo_path=${local_repo}/$(echo ${group_id}|tr '.' '/')/${artifact_id}/${version}/${qualified_jar}
+
+if test -f ${local_repo_path}; then
```

Comment:
neat trick here is that if you run maven deploy, it will be in the local repo. you don't need to play games this way

Reply:
heh actually we do.. in docker it won't see ~/.m2 in the context.. will have to re-introduce the arg to allow peeking in the cwd